src/main/java/com/ning/http/util/AsyncHttpProviderUtils.java

```diff
@@ -207,6 +207,19 @@ public static String getNTLM(List<String> authenticateHeaders) {
         return null;
     }
 
+    public static String getHttpHeaderForAuthScheme(List<String> authenticationHeaders, String authScheme)
+    {
+        for (String authenticateHeader : authenticationHeaders)
+        {
+            {
+                if (authenticateHeader.toUpperCase().startsWith(authScheme.toUpperCase())) {
```

Comment:
Since authScheme comes from an Enum, wouldn't it be uppercase already? Perhaps if you use `realm.getScheme().name()` instead of `toString()`.